src/astro/sql/operators/sql_decorator.py

```diff
@@ -95,23 +96,24 @@ def execute(self, context: Dict):
         self.run_id = context.get("run_id")
         self.convert_op_arg_dataframes()
         self.convert_op_kwarg_dataframes()
-        if not self.sql:
+        if self.sql is None or self.sql == "":
             sql_stuff = self.python_callable(*self.op_args, **self.op_kwargs)
             # If we return two things, assume the second thing is the params
             if len(sql_stuff) == 2:
                 self.sql, self.parameters = sql_stuff
             else:
                 self.sql = sql_stuff
                 self.parameters = {}
-        elif self.sql[-4:] == ".sql":
+        elif isinstance(self.sql, str) and self.sql[-4:] == ".sql":
```

Comment:
It feels weird to have `isinstance(self.sql, str)` a bunch fo times, maybe there should be one top level if statement and then a method inside of it?

Reply:
This is fixed now, removed the checks for string in this part of the code, added a check for string at starting of the sql_decorator's execute() method and split the code flow at that location for string and SQLAlchemy respectively.